virtual_ecosystem/models/litter/input_partition.py

```diff
@@ -0,0 +1,225 @@
+"""The ``models.litter.input_partition`` module handles the partitioning of dead plant
+and animal matter into the various pools of the litter model.
+"""  # noqa: D205
+
+import numpy as np
+from numpy.typing import NDArray
+
+from virtual_ecosystem.core.logger import LOGGER
+from virtual_ecosystem.models.litter.constants import LitterConsts
+
+# TODO - It makes sense for the animal pools to be handled here, but need to think about
+# how the partition works with the plant partition, Animals do not contain lignin, so if
+# I used the standard function on animal carcasses and excrement the maximum amount
+# (85%) will end up in the metabolic pool, which I think is basically fine, with bones
+# not being explicitly modelled I think this is fine. This will have to change once
+# bones are included.
+
+
+def partion_plant_inputs_between_pools(
+    deadwood_production: NDArray[np.float32],
+    leaf_turnover: NDArray[np.float32],
+    reproduct_turnover: NDArray[np.float32],
+    root_turnover: NDArray[np.float32],
+    leaf_turnover_lignin_proportion: NDArray[np.float32],
+    reproduct_turnover_lignin_proportion: NDArray[np.float32],
+    root_turnover_lignin_proportion: NDArray[np.float32],
+    leaf_turnover_c_n_ratio: NDArray[np.float32],
+    reproduct_turnover_c_n_ratio: NDArray[np.float32],
+    root_turnover_c_n_ratio: NDArray[np.float32],
+    constants: LitterConsts,
+) -> dict[str, NDArray[np.float32]]:
```

Comment:
Do you think this is a long signature? Have a look at [`pandas_read_csv`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas-read-csv). _That_ is long. 

Seriously, I don't see a big problem with having all of these inputs. The _look_ more than they are because the names are long (and descriptive, no complain in there), but otherwise it looks totally fine.